src/Core/Search/SearchPanelInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Search;
+
+interface SearchPanelInterface
+{
+    public function getTitle(): string;
+
+    public function getButtonLabel(): string;
+
+    public function buildLink(): string;
```

Comment:
I am not convinced by this name :thinking: 
I mean it is not the Interface business that your implementation as to "build" the link or not. IMO a `buildSomething` method should create an output from some inputs or a global / contextual state ...

I recommend `getLink` for the method name.

This comment is just a suggestion, not mandatory ^^

Reply:
I'm agree with your feedback but buildLink "builds" the link based on `link` and `queryParams` properties